test/process/io/PathSinkTest.cpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2019 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+const char *const PATH_SINK_TEST_SUITE = "Test suite for the PathSink class";
+const char *const PATH_SINK_CONSTRUCT = "Test construction and fulfillsPremises()";
+const char *const PATH_SINK_OPEN = "Test open and close";
+const char *const PATH_SINK_OPENLOCKED = "Test open and close with lock_unlock";
+const char *const PATH_SINK_AQUIRELOCK_LATER = "Test aquire lock after file open";
+const char *const PATH_SINK_WRITE_TELL_SEEK = "Test write tell seek rewind functions";
+const char *const PATH_SINK_WRITE_OVERWRITEBYTES = "Test write rewind_seek overwrite bytes";
+
+#include "process/io/PathSink.h"
+#include "process/io/PathSource.h"
+#include "TestConstants.h"
+#include "TestResourcesAndFunctions.h"
+#include <UnitTest++/UnitTest++.h>
+
+SUITE (PATH_SINK_TEST_SUITE) {
+    TEST (PATH_SINK_CONSTRUCT) {
+        TestResourcesAndFunctions res(PATH_SINK_TEST_SUITE, PATH_SINK_CONSTRUCT);
+
+        auto ps = res.filePath(TEST_FASTQ_SMALL);
+        PathSink p(ps);
+                CHECK(p.fulfillsPremises());
+                CHECK(p.getPath() == ps);
+                CHECK(!p.exists());
+                CHECK(!p.hasLock());
+                CHECK(!p.isOpen());
+                CHECK(p.isGood());
+                CHECK(!p.isBad());
+                CHECK(!p.eof());
+                CHECK(p.isFile());
```

Comment:
Either rename to `FileSink` (which I'd prefer) or to `isPath()`. See previous comment.